lib/expressions.js

```diff
@@ -60,21 +61,102 @@ class OperationExpression {
   interpret(context) {
     const args = this.operands.map((x) => x.interpret(context));
     const operator = OPERATORS[this.operator];
-    if (!operator) throw new Error(`Unknown operator: ${operator}`);
+    if (!operator) throw new Error(`Unknown operator: ${this.operator}`);
     if (this.operator.length > 1) return operator(...args);
     return args.reduce(operator);
   }
 
   toExpression() {
     const list = this.operands.map((x) => x.toExpression());
-    return '(' + list.join(this.operator) + ')';
+    return `(${list.join(this.operator)})`;
   }
 
   toJavaScript() {
-    const parameters = Array.from(this.identifiers);
-    const header = '(' + parameters.join(', ') + ')';
-    return header + ' => ' + this.toExpression();
+    let header = '(';
+
+    for (const identifier of this.identifiers) {
+      if (header.length > 1) {
+        header += ', ';
+      }
+      header += identifier;
+    }
+
+    header += ')';
+
+    return `${header} => ${this.toExpression()}`;
+  }
+}
+
+class ConditionExpression {
+  identifiers = new Set();
+
+  constructor(clauses) {
+    this.clauses = clauses;
+
+    for (const { condition, consequents } of clauses) {
+      if (condition.identifiers !== undefined) {
+        for (const identifier of condition.identifiers) {
+          this.identifiers.add(identifier);
+        }
+      }
+
+      for (const consequent of consequents) {
+        if (consequent.identifiers !== undefined) {
+          for (const identifier of consequent.identifiers) {
+            this.identifiers.add(identifier);
+          }
+        }
+      }
+    }
+  }
+
+  // eslint-disable-next-line consistent-return
+  interpret(context) {
+    for (const { condition, consequents } of this.clauses) {
+      if (condition.interpret(context) !== false) {
+        return consequents.reduce(
+          (_, consequent) => consequent.interpret(context),
+          undefined,
+        );
+      }
+    }
+  }
```

Comment:
Lisp returns nil if there are no matches. What behavior do we expect? Currently it returns undefined in this case